filters/accesslog/control.go

```diff
@@ -13,46 +13,70 @@ const (
 	AccessLogEnabledKey = "statebag:access_log:proxy:enabled"
 )
 
+// Common filter struct for holding access log state
+type AccessLogFilter struct {
+	Enable   bool
+	Prefixes []int
+}
+
+func (al *AccessLogFilter) Request(ctx filters.FilterContext) {
+	bag := ctx.StateBag()
+	bag[AccessLogEnabledKey] = al
+}
+
+func (*AccessLogFilter) Response(filters.FilterContext) {}
+
+func extractFilterValues(args []interface{}, enable bool) (filters.Filter, error) {
+	prefixes := make([]int, 0)
+	for _, prefix := range args {
+		intPref, ok := 0, false
+		switch prefix.(type) {
```

Comment:
I don't see why the switch case block is required.
Can't you always use `int()` ?
Maybe also `strconv.Atoi()` could be used as replacement.